discord/scheduled_events.go

```diff
@@ -0,0 +1,85 @@
+package discord
+
+import (
+	"time"
+
+	"github.com/diamondburned/arikawa/v3/utils/json/option"
+)
+
+// EventStatus describes the different statuses GuildScheduledEvent can be.
+//
+// https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-guild-scheduled-event-status
+type EventStatus int
+
+const (
+	Scheduled EventStatus = iota + 1
+	Active
+	Completed
+	Cancelled
+)
+
+// EntityType describes the different types GuildScheduledEvent can be.
+type EntityType int
+
+const (
+	StageInstanceEntity EntityType = iota + 1
+	VoiceEntity
+	ExternalEntity
+)
+
+// ScheduledEventPrivacy describes the privacy levels of GuildScheduledEvent.
+//
+// https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-guild-scheduled-event-privacy-level
+type ScheduledEventPrivacyLevel int
+
+const (
+	// GuildOnly requires the scheduled event to be only accessible to guild members.
+	GuildOnly ScheduledEventPrivacyLevel = iota + 2
+)
+
+// GuildScheduledEvent describes the scheduled event structure.
+//
+// https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-guild-scheduled-event-structure
+type GuildScheduledEvent struct {
+	// ID is the id of the scheduled event.
+	ID EventID `json:"id"`
+	// GuildID is the guild id of where the scheduled event belongs to.
+	GuildID GuildID `json:"guild_id"`
+	// ChannelID is the channel id in which the scheduled event will be
+	// hosted at, this may be NullChannelID if the EntityType is set
+	// to ExternalEntity.
+	ChannelID ChannelID `json:"channel_id"`
+	// CreatorID is the user id of who created the scheduled event.
+	CreatorID UserID `json:"creator_id"`
+	// Name is the name of the scheduled event.
+	Name string `json:"name"`
+	// Description is the description of the scheduled event.
+	Description option.NullableString `json:"description"`
```

Comment:
No difference between `null` and omitted here, so use a `string` with an `omitempty` flag.